src/Controller/CategoryAdminController.php

```diff
@@ -40,12 +40,13 @@ public static function getSubscribedServices(): array
 
     public function listAction(Request $request): Response
     {
-        if (null === $request->get('filter') && null === $request->get('filters')) {
+        if (null === $request->query->get('filter') && null === $request->query->get('filters')) {
             return new RedirectResponse($this->admin->generateUrl('tree', $request->query->all()));
         }
 
-        $listMode = $request->get('_list_mode');
+        $listMode = $request->query->get('_list_mode');
         if (null !== $listMode) {
+            /** @psalm-suppress InvalidArgument */
```

Comment:
Why gives invalid argument? maybe a string check is needed?

Reply:
Because the psalm-symfony plugin does completely understand the code, phpstan does.